app-starters-common-analytics/src/main/java/org/springframework/cloud/stream/app/metrics/AggregateCounterReader.java

```diff
@@ -59,4 +61,9 @@
 	 * @return an object containing an indexed array of the counts .
 	 */
 	AggregateCounter getCounts(String name, int nCounts, DateTime end, AggregateCounterResolution resolution);
+
+	/**
+	 * List the names of all available aggregate counters.
+	 */
+	Collection<String> list();
```

Comment:
I would vote for `List`. I would like to ensure that the returned Collection is sorted.